tools/gyp/pylib/gyp/generator/msvs.py

```diff
@@ -2171,6 +2171,9 @@ def _MapFileToMsBuildSourceType(source, rule_dependencies,
   elif ext == '.asm':
     group = 'masm'
     element = 'MASM'
+  elif ext == '.S':
```

Comment:
I'd suggest doing it like this
```suggestion
  elif ext.lower() in ['.asm', 's']:
```

Reply:
Is that equivalent? On Unices the convention is that .S should be run through cpp whereas .s should not. I don't know if that's applicable to usually-case-insensitive Windows.

Reply:
AFAICT `.s` nor `.S` are standard for MSVS. It's more common to see `.asm`.
And in this case while the file indeed has `.S` semantics on Unices, `cl.exe` doesn't except it.  

Seems like we have the same bug in `ninja.py`, so maybe the fix should be in `v8.gyp`, changing the extension to `.asm` iff Windows.

Reply:
https://github.com/nodejs/node/blob/0c64816c66f20ec78e4f620f98a774e94caea2ab/tools/gyp/pylib/gyp/generator/ninja.py#L1038-L1047